src/python/BuildingControlsSimulator/DataClients/DataChannel.py

```diff
@@ -25,13 +35,22 @@ def get_full_data_periods(self, expected_period):
                 self.spec.datetime_column, ascending=True
             )
             # drop records that are incomplete
+            original_len = len(self.data)
+            # TODO: can we overwrite self.data?
```

Comment:
Is this TODO still relevant?

Reply:
Yes, this will be refactored soon to not have so many copies of the input data.